lumicks/pylake/tests/test_imaging_mixins.py

```diff
@@ -96,47 +97,63 @@ def test_export_tiff_empty(tmp_path):
         export_empty.export_tiff(filename, dtype=np.float32)
 
 
-def test_export_tiff_int(tmp_path):
-    images = np.ones(shape=(2, 10, 10, 3))  # (n, h, w, c)
-    export_image16 = tiffexport_factory(images * np.iinfo(np.uint16).max)
-
-    # Sufficient bit-depth or forced clipping
-    export_image16.export_tiff(tmp_path / "uint16", dtype=np.uint16)
-    export_image16.export_tiff(tmp_path / "float32", dtype=np.float32)
-    export_image16.export_tiff(tmp_path / "clipped", dtype=np.uint8, clip=True)
-
-    # Raise because unsafe
-    with pytest.raises(RuntimeError, match="Can't safely export image with `dtype=uint8` channels"):
-        export_image16.export_tiff(tmp_path / "uint8", dtype=np.uint8)
+@pytest.mark.parametrize(
+    "output_dtype, data_dtype",
+    [
+        (np.float32, np.float32),
+        (np.float32, np.float16),
+        (np.float32, np.uint32),
+        (np.float16, np.float16),
+        (np.uint32, np.uint16),
+        (np.uint32, np.uint8),
+        (np.uint16, np.uint16),
+        (np.uint16, np.uint8),
+    ],
+)
+def test_export_tiff_sufficient_range(tmp_path, output_dtype, data_dtype):
+    """Test TIFF export when the data type has sufficient range for the data stored"""
+    info = np.finfo(data_dtype) if np.dtype(data_dtype).kind == "f" else np.iinfo(data_dtype)
+    images = np.ones(shape=(2, 10, 10, 3)) * info.max  # (n, h, w, c)
```

Comment:
Probably, one could make some/half of the pixels have the values of `info.min`?!

Reply:
I'll just change the first pixel to `info.min`.